File: machine_code_testing/speed_test.cpp

#include <iostream>
#include <vector>
#include <chrono>

int main() {
    auto start = std::chrono::high_resolution_clock::now(); // Start the timer
    std::vector<int> values;
    for (int i = 0; i < 100000; ++i) {
        values.push_back(i * i);
    }
    auto end = std::chrono::high_resolution_clock::now(); // End the timer
    auto duration = std::chrono::duration_cast<std::chrono::microseconds>(end - start);
    std::cout << "Elapsed time: " << duration.count() << " microseconds" << std::endl;
    return 0;
}

File: machine_code_testing/speed_test.py
import numba
import time

@numba.jit
def print_numbers():
    l = []
    for i in range(100000):
        l.append(i ** 2)

def time_function(func):
    start = time.time()
    func()
    end = time.time()
    print(f"Execution time: {(end - start) * 1e6:.0f} microseconds")

time_function(print_numbers)